src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -59,14 +59,11 @@ public T remove(int index) {
     @Override
     public T remove(T element) {
         for (int i = 0; i < size; i++) {
-            if (element == null) {
-                if (elements[i] == null) {
-                    return remove(i);
-                }
-            } else {
-                if (element.equals(elements[i])) {
-                    return remove(i);
-                }
+            if (element == null && elements[i] == null) {
+                return remove(i);
+            }
+            if (element != null && element.equals(elements[i])) {
+                return remove(i);
             }
```

Comment:
could you compose these two IFs into single one? The body is the same. You can move conditions to the separate methods if needed in order to make the code more clear